### DISCUSSION TITLE
Lesson Category-Based Lessons

### DISCUSSION THREAD

**Discussion by avinashtelsa** [Laravel-Backpack/community-forum#1197] [General]
Hello,

I need lessons based on the selection of lessonCategory, including all quotes for the lessons. Here is my code. Thank you for being so helpful.

![lessons](https://github.com/user-attachments/assets/06501813-afbd-4323-ace8-87312db07d7e)

protected function setupCreateOperation()
{
    Widget::add()->type('script')->content('js/admin/lesson_category_quotes.js');

    CRUD::setValidation(LessonCategoryQuotesRequest::class);
    CRUD::field('title');
    CRUD::field('course')
        ->type('select2')
        ->entity('course');
    CRUD::addField([
        'type'          => "relationship",
        'name'          => 'lessonCategory',
        'ajax'          => true,
        'minimum_input_length' => 0,
        'inline_create' => [ 'entity' => 'lesson-category' ] , // <--- THIS
        'data_source' => url('api/lesson-category'),
        'method' => 'GET',
    ]);
    CRUD::addField([
        'name'  => 'quotes',
        'label' => 'Quotes',
        'type'  => 'repeatable',
        'fields' => [
            [
                'name'  => 'title',
                'type'  => 'text',
                'label' => 'Quote Title',
            ],
            [
                'name'  => 'lessons',
                'type'  => 'select2_multiple',
                'label' => 'Lessons',
                'entity' => 'lessons',
                'model' => 'App\Models\Lesson',
                'attribute' => 'title',
                'pivot' => true,
                'default' => null,
                'placeholder' => 'Select lessons',
                'attributes' => [
                    'class' => 'lesson-select',
                ],
            ],
        ],
        'new_item_label'  => 'Add Quote',
        'init_rows'       => 0,
        'min_rows'        => 0,
        'max_rows'        => 10,
    ]);

    /**
        * Fields can be defined using the fluent syntax:
        * - CRUD::field('price')->type('number');
        */
}

lesson_category_quotes.js

crud.field('lessonCategory').onChange(function (field) {
    let selectedCategoryId = field.value;

    console.log('Selected Category ID:', selectedCategoryId);

    if (selectedCategoryId) {
        $.ajax({
            url: 'http://api.lightup-tutoring.local/api/get-lessons-by-category/' + selectedCategoryId,
            method: 'GET',
            success: function (data) {
                console.log('Lessons fetched:', data);
            },
            error: function (err) {
                console.error('Error fetching lessons:', err);
            }
        });
    }
});

**Comment by jcastroa87** [comment]
Hello @avinashtelsa 

You can try with [select2_from_ajax_multiple](https://backpackforlaravel.com/docs/6.x/crud-fields#select2_from_ajax_multiple-pro) with "include_all_form_fields" and handle from Controller search function.

In any case, need to think in a way if the user change lessonCategory, what you will do with all repeatable content.

Cheers.

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
